src/cephadm/cephadm

```diff
@@ -3589,14 +3589,7 @@ def prepare_ssh(
     cli: Callable, wait_for_mgr_restart: Callable
 ) -> None:
 
-    cli(['config-key', 'set', 'mgr/cephadm/ssh_user', ctx.ssh_user])
-
-    logger.info('Enabling cephadm module...')
-    cli(['mgr', 'module', 'enable', 'cephadm'])
-    wait_for_mgr_restart()
-
-    logger.info('Setting orchestrator backend to cephadm...')
-    cli(['orch', 'set', 'backend', 'cephadm'])
+    cli(['cephadm', 'set-ssh-user', ctx.ssh_user])
```

Comment:
```
2021-04-05T07:35:26.808 INFO:journalctl@ceph.mon.a.gibba005.stdout:Apr 05 07:35:26 gibba005 conmon[28926]: mgr.14116) 2 : audit [DBG] from='client.14120 -' entity='client.admin' cmd=[{"prefix": "mgr_status"}]: dispatch
2021-04-05T07:35:27.314 INFO:teuthology.orchestra.run.gibba005.stderr:Running command: /bin/podman run --rm --ipc=host --no-hosts --net=host --entrypoint /usr/bin/ceph --init -e CONTAINER_IMAGE=quay.ceph.io/ceph-ci/ceph:95a8e3ead69b8bd2139389ff9c333b41ac934972 -e NODE_NAM
E=gibba005 -e CEPH_USE_RANDOM_NONCE=1 -v /var/log/ceph/559a4eda-95e1-11eb-8186-001a4aab830c:/var/log/ceph:z -v /tmp/ceph-tmpiwsh7zrs:/etc/ceph/ceph.client.admin.keyring:z -v /tmp/ceph-tmp1eykleiu:/etc/ceph/ceph.conf:z quay.ceph.io/ceph-ci/ceph:95a8e3ead69b8bd2139389ff9c33
3b41ac934972 cephadm set-ssh-user root
2021-04-05T07:35:27.896 INFO:teuthology.orchestra.run.gibba005.stderr:/usr/bin/ceph: no valid command found; 10 closest matches:
2021-04-05T07:35:27.897 INFO:teuthology.orchestra.run.gibba005.stderr:/usr/bin/ceph: cephadm set-ssh-config
2021-04-05T07:35:27.897 INFO:teuthology.orchestra.run.gibba005.stderr:/usr/bin/ceph: cephadm clear-ssh-config
2021-04-05T07:35:27.898 INFO:teuthology.orchestra.run.gibba005.stderr:/usr/bin/ceph: cephadm get-ssh-config
2021-04-05T07:35:27.898 INFO:teuthology.orchestra.run.gibba005.stderr:/usr/bin/ceph: cephadm generate-key
2021-04-05T07:35:27.899 INFO:teuthology.orchestra.run.gibba005.stderr:/usr/bin/ceph: cephadm set-priv-key
2021-04-05T07:35:27.899 INFO:teuthology.orchestra.run.gibba005.stderr:/usr/bin/ceph: cephadm set-pub-key
2021-04-05T07:35:27.900 INFO:teuthology.orchestra.run.gibba005.stderr:/usr/bin/ceph: cephadm clear-key
2021-04-05T07:35:27.901 INFO:teuthology.orchestra.run.gibba005.stderr:/usr/bin/ceph: cephadm get-pub-key
2021-04-05T07:35:27.901 INFO:teuthology.orchestra.run.gibba005.stderr:/usr/bin/ceph: cephadm get-user
2021-04-05T07:35:27.902 INFO:teuthology.orchestra.run.gibba005.stderr:/usr/bin/ceph: cephadm set-user <user>
2021-04-05T07:35:27.902 INFO:teuthology.orchestra.run.gibba005.stderr:/usr/bin/ceph: Error EINVAL: invalid command
2021-04-05T07:35:27.959 INFO:journalctl@ceph.mon.a.gibba005.stdout:Apr 05 07:35:27 gibba005 conmon[28926]: cluster 2021-04-05T07:35:26.694754+0000 mon.a (mon.0) 49 : cluster [DBG] mgrmap e8: y(active, since 2s)
2021-04-05T07:35:27.960 INFO:journalctl@ceph.mon.a.gibba005.stdout:Apr 05 07:35:27 gibba005 conmon[28926]: audit 2021-04-05T07:35:26.730424+0000 mgr.y (mgr.14116) 3 : audit [DBG] from='client.14128 -' entity='client.admin' cmd=[{"prefix": "orch set backend", "module_name"
: "cephadm", "target": ["mon-mgr", ""]}]: dispatch
2021-04-05T07:35:27.960 INFO:journalctl@ceph.mon.a.gibba005.stdout:Apr 05 07:35:27 gibba005 conmon[28926]: audit 2021-04-05T07:35:26.733153+0000 mon.a (mon.0) 50 : audit [INF] from='mgr.14116 172.21.2.105:0/3988907738' entity='mgr.y'
2021-04-05T07:35:27.960 INFO:journalctl@ceph.mon.a.gibba005.stdout:Apr 05 07:35:27 gibba005 conmon[28926]: audit 2021-04-05T07:35:26.736130+0000 mon.a (mon.0) 51 : audit [DBG] from='mgr.14116 172.21.2.105:0/3988907738' entity='mgr.y' cmd=[{"prefix": "config dump", "format
": "json"}]: dispatch
2021-04-05T07:35:28.457 INFO:teuthology.orchestra.run.gibba005.stderr:Non-zero exit code 22 from /bin/podman run --rm --ipc=host --no-hosts --net=host --entrypoint /usr/bin/ceph --init -e CONTAINER_IMAGE=quay.ceph.io/ceph-ci/ceph:95a8e3ead69b8bd2139389ff9c333b41ac934972 -
e NODE_NAME=gibba005 -e CEPH_USE_RANDOM_NONCE=1 -v /var/log/ceph/559a4eda-95e1-11eb-8186-001a4aab830c:/var/log/ceph:z -v /tmp/ceph-tmpiwsh7zrs:/etc/ceph/ceph.client.admin.keyring:z -v /tmp/ceph-tmp1eykleiu:/etc/ceph/ceph.conf:z quay.ceph.io/ceph-ci/ceph:95a8e3ead69b8bd213
9389ff9c333b41ac934972 cephadm set-ssh-user root
2021-04-05T07:35:28.457 INFO:teuthology.orchestra.run.gibba005.stderr:/usr/bin/ceph: stderr no valid command found; 10 closest matches:
2021-04-05T07:35:28.458 INFO:teuthology.orchestra.run.gibba005.stderr:/usr/bin/ceph: stderr cephadm set-ssh-config
2021-04-05T07:35:28.458 INFO:teuthology.orchestra.run.gibba005.stderr:/usr/bin/ceph: stderr cephadm clear-ssh-config
2021-04-05T07:35:28.458 INFO:teuthology.orchestra.run.gibba005.stderr:/usr/bin/ceph: stderr cephadm get-ssh-config
2021-04-05T07:35:28.458 INFO:teuthology.orchestra.run.gibba005.stderr:/usr/bin/ceph: stderr cephadm generate-key
2021-04-05T07:35:28.459 INFO:teuthology.orchestra.run.gibba005.stderr:/usr/bin/ceph: stderr cephadm set-priv-key
2021-04-05T07:35:28.459 INFO:teuthology.orchestra.run.gibba005.stderr:/usr/bin/ceph: stderr cephadm set-pub-key
2021-04-05T07:35:28.459 INFO:teuthology.orchestra.run.gibba005.stderr:/usr/bin/ceph: stderr cephadm clear-key
2021-04-05T07:35:28.459 INFO:teuthology.orchestra.run.gibba005.stderr:/usr/bin/ceph: stderr cephadm get-pub-key
2021-04-05T07:35:28.459 INFO:teuthology.orchestra.run.gibba005.stderr:/usr/bin/ceph: stderr cephadm get-user
2021-04-05T07:35:28.460 INFO:teuthology.orchestra.run.gibba005.stderr:/usr/bin/ceph: stderr cephadm set-user <user>
2021-04-05T07:35:28.460 INFO:teuthology.orchestra.run.gibba005.stderr:/usr/bin/ceph: stderr Error EINVAL: invalid command
2021-04-05T07:35:28.460 INFO:teuthology.orchestra.run.gibba005.stderr:Traceback (most recent call last):
2021-04-05T07:35:28.461 INFO:teuthology.orchestra.run.gibba005.stderr:  File "/home/ubuntu/cephtest/cephadm", line 7933, in <module>
2021-04-05T07:35:28.461 INFO:teuthology.orchestra.run.gibba005.stderr:    main()
2021-04-05T07:35:28.461 INFO:teuthology.orchestra.run.gibba005.stderr:  File "/home/ubuntu/cephtest/cephadm", line 7921, in main
2021-04-05T07:35:28.462 INFO:teuthology.orchestra.run.gibba005.stderr:    r = ctx.func(ctx)
2021-04-05T07:35:28.462 INFO:teuthology.orchestra.run.gibba005.stderr:  File "/home/ubuntu/cephtest/cephadm", line 1717, in _default_image
2021-04-05T07:35:28.462 INFO:teuthology.orchestra.run.gibba005.stderr:    return func(ctx)
2021-04-05T07:35:28.462 INFO:teuthology.orchestra.run.gibba005.stderr:  File "/home/ubuntu/cephtest/cephadm", line 3992, in command_bootstrap
2021-04-05T07:35:28.463 INFO:teuthology.orchestra.run.gibba005.stderr:    prepare_ssh(ctx, cli, wait_for_mgr_restart)
2021-04-05T07:35:28.463 INFO:teuthology.orchestra.run.gibba005.stderr:  File "/home/ubuntu/cephtest/cephadm", line 3605, in prepare_ssh
2021-04-05T07:35:28.463 INFO:teuthology.orchestra.run.gibba005.stderr:    cli(['cephadm', 'set-ssh-user', ctx.ssh_user])
2021-04-05T07:35:28.463 INFO:teuthology.orchestra.run.gibba005.stderr:  File "/home/ubuntu/cephtest/cephadm", line 3936, in cli
2021-04-05T07:35:28.463 INFO:teuthology.orchestra.run.gibba005.stderr:    ).run(timeout=timeout)
2021-04-05T07:35:28.464 INFO:teuthology.orchestra.run.gibba005.stderr:  File "/home/ubuntu/cephtest/cephadm", line 3174, in run
2021-04-05T07:35:28.464 INFO:teuthology.orchestra.run.gibba005.stderr:    desc=self.entrypoint, timeout=timeout)
2021-04-05T07:35:28.464 INFO:teuthology.orchestra.run.gibba005.stderr:  File "/home/ubuntu/cephtest/cephadm", line 1411, in call_throws
2021-04-05T07:35:28.464 INFO:teuthology.orchestra.run.gibba005.stderr:    raise RuntimeError('Failed command: %s' % ' '.join(command))
2021-04-05T07:35:28.465 INFO:teuthology.orchestra.run.gibba005.stderr:RuntimeError: Failed command: /bin/podman run --rm --ipc=host --no-hosts --net=host --entrypoint /usr/bin/ceph --init -e CONTAINER_IMAGE=quay.ceph.io/ceph-ci/ceph:95a8e3ead69b8bd2139389ff9c333b41ac93497
2 -e NODE_NAME=gibba005 -e CEPH_USE_RANDOM_NONCE=1 -v /var/log/ceph/559a4eda-95e1-11eb-8186-001a4aab830c:/var/log/ceph:z -v /tmp/ceph-tmpiwsh7zrs:/etc/ceph/ceph.client.admin.keyring:z -v /tmp/ceph-tmp1eykleiu:/etc/ceph/ceph.conf:z quay.ceph.io/ceph-ci/ceph:95a8e3ead69b8bd
2139389ff9c333b41ac934972 cephadm set-ssh-user root
```

https://pulpito.ceph.com/kchai-2021-04-05_06:39:05-rados-wip-kefu-testing-2021-04-05-1234-distro-basic-gibba/

Reply:
This should work now, can you test it again?

Reply:
i am adding the 'needs-qa' label, so some of us can pick it up when testing.